File: backend/src/main/scala/review/api/SubmitReviewAppealApi.scala
package review.api

import domain.shared.given

import cats.effect.IO
import domain.review.ReviewAppealRequest
import domain.shared.{AppealRole, DeliveryAppState, OrderId, UserRole}
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import review.app.submitReviewAppeal
import shared.api.routing.*
import shared.app.*

val submitReviewAppealApi: FixedMethodApi1[OrderId, DeliveryAppState] =
  jsonPostApi1[OrderId, ReviewAppealRequest, DeliveryAppState](
    List(routeSegment("api"), routeSegment("delivery"), routeSegment("orders")),
    List(routeSegment("review-appeals")),
  )

File: frontend/src/apis/review/SubmitReviewAppealApi.ts
import type { DeliveryAppState, OrderId, ReviewAppealRequest } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { buildApiPath1, defineJsonPostApi1, routeSegment } from '@/system/api/TypedApiDefinitions'

export const submitReviewAppealApiDefinition = defineJsonPostApi1<OrderId, ReviewAppealRequest, DeliveryAppState>(
    [routeSegment('api'), routeSegment('delivery'), routeSegment('orders')],
    [routeSegment('review-appeals')],
  )

export function submitReviewAppeal(orderId: OrderId, payload: ReviewAppealRequest) {
  return postNormalizedDeliveryState(buildApiPath1(submitReviewAppealApiDefinition, orderId), payload)
}
